xamarin/Xamarin.Forms
# Add Brushes support to Shapes

### Description of Change ###

Add Brushes support to Shapes.

_NOTE: This PR is a draft temporarily. The PR is based on https://github.com/xamarin/Xamarin.Forms/pull/9220. After merge the 9220 changes, this PR changes will be reduced to a dozen files (much less than the current ones)._

### API Changes ###
 
In all Shapes (Ellipse, Rectangle, Line, etc.) changes:
- Fill property (Color to Brush)
- Stroke property (Color to Brush)

### Platforms Affected ### 
- Core/XAML (all platforms)
- iOS
- macOS
- Android
- UWP
- WPF

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

![shapes-brushes](https://user-images.githubusercontent.com/6755973/84171191-d7b64480-aa7a-11ea-8d92-8471a91a345c.gif)
<img width="1916" alt="shapes-brushes-wpf" src="https://user-images.githubusercontent.com/6755973/84171237-e3a20680-aa7a-11ea-9eac-5ee1b9f740d6.png">
<img width="1351" alt="shapes-brushes-uwp" src="https://user-images.githubusercontent.com/6755973/84171238-e43a9d00-aa7a-11ea-9c3e-2a748072d601.png">

### Testing Procedure ###
Launch Core Gallery and navigate to the Brush Gallery. Open the "Shapes Brush Gallery" and verify that brushes work correctly with all shapes.

### PR Checklist ###
